rotkehlchen/tests/unit/decoders/test_thegraph.py

```diff
@@ -93,6 +95,158 @@ def test_thegraph_delegate(database, ethereum_inquirer):
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1706311139000)
+    gas, deposit_amount = '0.000626151499903872', '0.001135647343563552'
+    contract = string_to_evm_address('0xCa82c7Ce3388b0B5d307574099aC57d7a00d509F')
+    expected_events = [
+        EvmEvent(
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas)),
+            location_label=ADDY_ROTKI,
+            notes=f'Burned {gas} ETH for gas',
+            tx_hash=tx_hash,
+            counterparty=CPT_GAS,
+        ),
+        EvmEvent(
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(deposit_amount), usd_value=FVal(0)),
+            location_label=ADDY_ROTKI,
+            notes=f'Deposit {deposit_amount} ETH to {contract} contract to pay for the gas in L2.',
+            tx_hash=tx_hash,
+            counterparty=CPT_THEGRAPH,
+            address=contract,
+        ),
+    ]
+    assert expected_events == events
+
+
+@pytest.mark.parametrize('ethereum_accounts', [[ADDY_ROTKI]])
+def test_thegraph_contract_delegation_transferred_to_l2_vested(database, ethereum_inquirer):
+    tx_hash = deserialize_evm_tx_hash('0x48321bb00e5c5b67f080991864606dbc493051d20712735a579d7ae31eca3d78')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1706316575000)
+    gas, delegation_amount = '0.0034549683606123', '199846.719749385820105919'
+    indexer = string_to_evm_address('0x5A8904be09625965d9AEc4BFfD30D853438a053e')
+    indexer_l2 = string_to_evm_address('0x2f09092aacd80196FC984908c5A9a7aB3ee4f1CE')
+    delegator_l2 = string_to_evm_address('0x9F219c3D048967990f675F49C1117B0598331408')
+    contract = string_to_evm_address('0x7D91717579885BfCFec3Cb4B4C4fe71c1EedD4dE')
+    expected_events = [
+        EvmEvent(
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas)),
+            location_label=ADDY_ROTKI,
+            notes=f'Burned {gas} ETH for gas',
+            identifier=None,
+            tx_hash=tx_hash,
+            counterparty=CPT_GAS,
+        ),
+        EvmEvent(
```

Comment:
In this one and the others in this file
```suggestion
        ), EvmEvent(
```